fhirr4/ballerina/src/main/resources/fhirservice/http_service_builder.bal

```diff
@@ -15,23 +15,32 @@
 // under the License.
 
 import ballerina/http;
+import ballerina/regex;
 import ballerinax/health.fhir.r4;
 
 // Construct an http service for a fhir service.
-isolated function getHttpService(Holder h, r4:ResourceAPIConfig apiConfig) returns http:Service {
+isolated function getHttpService(Holder h, r4:ResourceAPIConfig apiConfig, string[] & readonly name) returns http:Service {
     http:InterceptableService httpService = isolated service object {
 
         private final Holder holder = h;
         private final FHIRPreprocessor preprocessor = new (apiConfig);
+        private string[] name = name;
 
         public function createInterceptors() returns [FHIRResponseErrorInterceptor, FHIRResponseInterceptor] {
             return [new FHIRResponseErrorInterceptor(), new FHIRResponseInterceptor(apiConfig)];
         }
 
-        isolated resource function get [string... paths](http:Request req, http:RequestContext ctx) returns any|error {
+        isolated resource function get [string... path](http:Request req, http:RequestContext ctx) returns any|error { 
 
+            string[] rawPath = regex:split(req.rawPath, "\\?");
+            string[] paths = regex:split(rawPath[0], "/");
```

Comment:
regex is deprecated and less performant. use regexp:split instead ( https://central.ballerina.io/ballerina/lang.regexp/2201.7.0#split)